### PR TITLE
[TASK] Revise Backend JavaScript modal section

### PR DESCRIPTION
**Pull request by linawolf**
In preparation for https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/164

releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 1]
```diff
@@ -10,13 +10,16 @@ Modals
...
+TYPO3 provides an API as basis to create modal windows with severity
+representation. For better UX, if actions (buttons) are attached to the modal,
+one button must be a positive action. This button should get a `btnClass`
+to highlight it.
 
-Modals should be used rarely and only for confirmations. For information the :code:`TYPO3.Flashmessage` API should be used.
-For complex content, like forms or a lot of information, please use normal pages.
+Modals should be used rarely and only for confirmations. For information that
+does not require a confirmation
+the :ref:`Notification API (flash message) <notification_api>` API should be used.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:20]:
```suggestion
the :ref:`Notification API (flash message) <notification_api>` should be used.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 2]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+    :Required: true
+    :type: string|jQuery
+
+    The content displayed in the modal
+
+..  confval:: severity
+
+    :type: int
+    :Default: :code:`TYPO3.Severity.info`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:53]:
```suggestion
    :Default: :js:`TYPO3.Severity.info`
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 3]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+    Represents the severity of a modal. Please see :code:`TYPO3.Severity`.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:55]:
```suggestion
    Represents the severity of a modal. Please see :js:`TYPO3.Severity`.
```

Please see -> can we link it?

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 4]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+..  confval:: buttons
+
+    :type: object[]
+
+    Actions rendered into the modal footer. If empty, the footer
+    is not rendered. See table below.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:62]: There is no table below. Can we link to the next section?

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 5]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+    The text rendered into the button.
+
+..  confval:: trigger / action
+
+    :Required: true
+    :type: function
+
+    Callback that is triggered on button click - either simple function or
+    `DeferredAction` / `ImmediateAction`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:81]:
```suggestion
    :js:`DeferredAction` / :js:`ImmediateAction`
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 6]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+    :type: string
+
+    The text rendered into the button.
+
+..  confval:: trigger / action
+
+    :Required: true
+    :type: function
+
+    Callback that is triggered on button click - either simple function or
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:80]:
```suggestion
    Callback that is triggered on button click - either a simple function or
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 7]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+    :type: bool
+
+    Marks the button as active. If true, the button gets the focus.
+
+..  confval:: btnClass
+
+    :type: string
+
+    The css class for the button
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:93]:
```suggestion
    The CSS class for the button.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 8]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
 
 
 Data Attributes
 ---------------
 
 It is also possible to use data-attributes to trigger a modal.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:99]:
```suggestion
It is also possible to use :html:`data` attributes to trigger a modal,
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 9]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+for example on an anchor element, which prevents the default behavior.
+
+:html:`data-title`
+    the title text for the modal
+
+:html:`data-bs-content`
+    the content text for the modal
+
+:html:`data-severity`
+    the severity for the modal, default is info (see :code:`TYPO3.Severity.*`)
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:109]:
```suggestion
    the severity for the modal, default is info (see :js:`TYPO3.Severity.*`)
```

Can we link TYPO3.Security?

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 10]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
-========================= ===========
-data-title                the title text for the modal
-data-bs-content              the content text for the modal
-data-severity             the severity for the modal, default is info (see :code:`TYPO3.Severity.*`)
-data-href                 the target URL, default is the href attribute of the element
-data-button-close-text    button text for the close/cancel button
-data-button-ok-text       button text for the ok button
-========================= ===========
+:html:`data-href`
+    the target URL, default is the href attribute of the element
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:112]:
```suggestion
    The target URL, default is the :html:`href` attribute of the element.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 11]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
 
 Data Attributes
 ---------------
 
 It is also possible to use data-attributes to trigger a modal.
-e.g. on an anchor element, which prevents the default behavior.
+for example on an anchor element, which prevents the default behavior.
+
+:html:`data-title`
+    the title text for the modal
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:103]:
```suggestion
    The title text for the modal.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 12]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+:html:`data-bs-content`
+    the content text for the modal
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:106]:
```suggestion
    The content text for the modal.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst (hunk 9 above)
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:109]:
```suggestion
    The severity for the modal, default is `info` (see :code:`TYPO3.Severity.*`).
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 13]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
-data-href                 the target URL, default is the href attribute of the element
-data-button-close-text    button text for the close/cancel button
-data-button-ok-text       button text for the ok button
-========================= ===========
+:html:`data-href`
+    the target URL, default is the href attribute of the element
 
-:code:`class="t3js-modal-trigger"` marks the element as modal trigger
+:html:`data-button-close-text`
+    button text for the close/cancel button
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:115]:
```suggestion
    Button text for the :guilabel:`close`/:guilabel:`cancel` button.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 14]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
 
+:html:`data-button-ok-text `
+    button text for the ok button
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:118]:
```suggestion
    Button text for the :guilabel:`ok` button.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 15]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+
+:html:`class="t3js-modal-trigger"`
+    marks the element as modal trigger
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:121]:
```suggestion
    Marks the element as modal trigger.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/_Modals/_DataModal.html [hunk 16]
```diff
@@ -0,0 +1,7 @@
+<a
+    href="delete.php"
+    class="t3js-modal-trigger"
+    data-title="Delete"
+    data-bs-content="Really delete?">
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/_Modals/_DataModal.html:5]:
```suggestion
    data-bs-content="Really delete?"
>
```

Had to look twice where it ends.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:55]: Not sure where to link this to?

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 17]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
 #. :code:`TYPO3.Modal.dismiss()`
```
Comment (linawolf) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:30]:
```suggestion
#. :js:`TYPO3.Modal.dismiss()`
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 18]
```diff
@@ -27,91 +30,118 @@ The API provides only two public methods:
...
+for example on an anchor element, which prevents the default behavior.
+
+:html:`data-title`
+    The title text for the modal.
+
+:html:`data-bs-content`
+    The content text for the modal.
+
+:html:`data-severity`
+    The severity for the modal, default is `info` (see :code:`TYPO3.Severity.*`).
```
Comment (linawolf) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:109]:
```suggestion
    The severity for the modal, default is `info` (see :js:`TYPO3.Severity.*`).
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 19]
```diff
@@ -10,108 +10,138 @@ Modals
...
+the :ref:`Notification API (flash message) <notification_api>` should be used.
 
+For complex content, like forms or a lot of information, use normal pages.
 
 API
 ===
 
 The API provides only two public methods:
 
 #. :code:`TYPO3.Modal.confirm(title, content, severity, buttons)`
```
Comment (linawolf) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:29]:
```suggestion
#. :js:`TYPO3.Modal.confirm(title, content, severity, buttons)`
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 20]
```diff
@@ -10,108 +10,138 @@ Modals
...
 
 API
 ===
 
 The API provides only two public methods:
 
-#. :code:`TYPO3.Modal.confirm(title, content, severity, buttons)`
-#. :code:`TYPO3.Modal.dismiss()`
+#. :js:`TYPO3.Modal.confirm(title, content, severity, buttons)`
+#. :js:`TYPO3.Modal.dismiss()`
```
Comment (linawolf) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:32]:
```suggestion
.. _modules-modals-settings:

```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 21]
```diff
@@ -10,108 +10,138 @@ Modals
...
+    Represents the severity of a modal. Please see :js:`TYPO3.Severity`.
+
+..  confval:: buttons
+
+    :type: object[]
+
+    Actions rendered into the modal footer. If empty, the footer
+    is not rendered. See table below.
+
+
```
Comment (linawolf) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:64]:
```suggestion
.. _modules-modals-button-settings:

```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst [hunk 22]
```diff
@@ -10,108 +10,140 @@ Modals
...
+    :Default: :js:`TYPO3.Severity.info`
+
+    Represents the severity of a modal. Please see :js:`TYPO3.Severity`.
+
+..  confval:: buttons
+
+    :type: object[]
+
+    Actions rendered into the modal footer. If empty, the footer
+    is not rendered. See table below.
```
Comment (linawolf) [inline Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst:63]:
```suggestion
    is not rendered. See section :ref:`modules-modals` on how to configure the buttons.
```

File: Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/ApiOverview/Backend/JavaScript/Modules/Modals.rst]: done

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**